text/0510-dynamodb-global-table.md

```diff
@@ -0,0 +1,1742 @@
+ either table name, table arn or table attributes to import it.
+
+```typescript
+GlobalTable.fromTableName(stack, 'FooTableId', 'FooTable');
+```
+
+### NOTE
+
+* Be careful while naming your global table. If there is a name conflict in a region with a DynamoDB table,
+that could lead to removal of that table with the global table replica.
+* We only support version [2019.11.21](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/globaltables.V2.html)
+of the global table since CloudFormation just has support for the same.
+
+---
+
+Ticking the box below indicates that the public API of this RFC has been
+signed-off by the API bar raiser (the `api-approved` label was applied to the
+RFC pull request):
+
+```
+[ ] Signed-off by API Bar Raiser @xxxxx
+```
+
+## Public FAQ
+
+### What are we launching today?
+
+We are launching L2 support for DynamoDB Global Table feature.
+
+### Why should I use this feature?
+
+You should use this feature if you would like to provision a DynamoDB table that has the capability
+of replicating to other regions. This would remove the need of creating custom solution for replicating
+your table to other regions and decrease maintenance load.
+
+## Internal FAQ
+
+### Why are we doing this?
+
+DynamoDB Global table L2 support has been requested by our users for a long time. When CDK initally added
+support for this feature, CloudFormation support for global tables did not exist. So we had to use
+[custom CloudFormation resources](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.custom_resources-readme.html)
+within our [Table construct](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_dynamodb-readme.html#amazon-dynamodb-global-tables)
+to add support for this feature.
+
+The current implementation has some limitations,
+
+* Some properties are not propagated across replicas.
+  * Open Issues: #25740, #25443, #18582
+* Customer managed key is not supported with replicationRegions in current solution.
+(Issues: #15957)
+
+The current solution also would add maintenance load and cost for custom resource in user stack.
+And also blocks users who do not want to use a custom resource solution for provisioning their global tables.
+
+This implementation moves away from the custom resource solution and to the user experience that
+global table intented to provide. With CloudFormation now having support for global table resource,
+we can now add L2 support for this feature.
+
+### Why should we _not_ do this?
+
+We currently offer two solution to provision a global table,
+
+1. Custom CloudFormation Resource within Table construct
+2. L1 for global table resource
+
+This means the customer is not blocked to use global tables. Adding L2 support would take up developer
+time and effort and, would also add to maintenance load for the CDK team.
+
+### What is the technical solution (design) of this feature?
```

Comment:
Please focus on the high level decisions, no need to paste entire source files in here.

The most interesting to me are:

- The design of specifying replicas itself
- Throughput specifications for table, replicas, GSIs
- Per-replica KMS keys (because of the nature of those KMS keys).
- How tables are referenced/used in downstream (replicated) stacks.

 Please include realistic examples that have multiple stacks that deploy to different regions.